### PR TITLE
implement GET user endpoint

### LINKED ISSUE #18: Username vs Loginname
As the Nextcloud login flow [docs](https://docs.nextcloud.com/server/latest/developer_manual/client_apis/LoginFlow/index.html) mention there might be cases where the login name is not the same as the username needed for the Webdav URL. Therefor it is necessary to provide access to the `<server>/ocs/v1.php/cloud/user` OCS API endpoint to be able to fetch the real username.

This process could be enclosed in the library itself in form of a login method or made available just as a simple call like everything else.

### PR DESCRIPTION
**Pull request by vauvenal5**
closes #18

### CODE REVIEW

File: lib/src/user/userdata.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  // ignore: public_member_api_docs
+  UserData(
+    this.id,
+    this.displayName,
+    this.email,
+    this.storageLocation,
+  );
+
+  // ignore: public_member_api_docs
+  factory UserData.fromJson(Map<String, dynamic> json) => UserData(
```
Comment (vauvenal5) [inline lib/src/user/userdata.dart:12]: Wouldn't an object mapper library like [dart_josn_mapper](https://pub.dev/packages/dart_json_mapper) come in handy for this?

Just a thought for the future.

Comment (provokateurin) [reply lib/src/user/userdata.dart:12]: Sounds like a good idea.

**Comment by provokateurin** [comment]
I can't test this right now, because I have no Nextcloud instance. I'll implement https://github.com/Viktoriaschule/dart-nextcloud/issues/16 first and then review this, ok?

**Comment by fingeg** [comment]
I have an active instance and tested your PR. Everything works fine.
Thanks for contributing.

**Comment by provokateurin** [comment]
Published!

**Pull request merged**